src/kolibri_android/kolibri_utils.py

```diff
@@ -0,0 +1,139 @@
+import logging
+import os
+import re
+from importlib.util import find_spec
+
+from .globals import SCRIPT_PATH
+
+
+# These Kolibri plugins conflict with the plugins listed in REQUIRED_PLUGINS
+# or OPTIONAL_PLUGINS:
+DISABLED_PLUGINS = [
+    "kolibri.plugins.learn",
+]
+
+# These Kolibri plugins must be enabled for the application to function
+# correctly:
+REQUIRED_PLUGINS = [
+    "kolibri.plugins.app",
+]
+
+# These Kolibri plugins will be dynamically enabled if they are available:
+OPTIONAL_PLUGINS = [
+    "kolibri_explore_plugin",
+    "kolibri_zim_plugin",
+]
+
+
+def init_kolibri(content_fallback_dirs=None, **kwargs):
+    logging.info("Initializing Kolibri and running any upgrade routines")
+
+    _monkeypatch_whitenoise()
+    _init_kolibri_env(content_fallback_dirs)
+
+    from kolibri.utils.main import initialize
```

Comment:
One more question. I see a lot of imports inside functions in the `kolibri_utils.py` file. However the Python PEP8 recommends to put all imports at the top of the file.

Reply:
Yeah, these are to deal with Kolibri imports that require various environment setup to happen first. In this case, importing `initialize` also means importing `conf`, which does a lot of things on import.

Probably worth going through them again and making sure it only happens where it's absolutely necessary. Alternatively, could do lazy imports to make things nicer.

But that bit in particular I'll fix right now. Imports in the middle of a function are _extra_ bad :)

Reply:
Usually these kind of lazy importing are an indicator that the module needs to be split further. What do you think about doing that?

Reply:
Yeah, I'd like to get things to a state where we're working within Kolibri's world such that it doesn't _matter_ if importing has side-effects. We should be configuring our environment by setting things in `kolibri.conf` - which we can do any time before running `initialize()` - not by using environment variables.